src/docs/transaction-clustering/index.mdx

```diff
@@ -0,0 +1,147 @@
+   1. writes the rules into project options.
+1. The derived rules are written into the project config submitted to Relay.
+1. For every incoming transaction of type URL, Relay applies matching rules to scrub identifiers.
+
+### Rule TTL
+
+In order to prune unused rules, we keep a copy of discovered rules in redis, and in post processing
+bump a `last_used` field on a rule if it was applied to the current event.
+The information which rule was applied is provided by Relay through the `_meta` field.
+
+## Known Issues
+
+### Accidental erasure of non-identifiers
+
+Every level of the tree of URL path segments could contain a mixture of identifiers and static pages, for example
+
+```
+
+/user/alice
+/user/bob
+/user/settings <-
+/user/carol
+...
+
+```
+
+our current approach would replace `/user/settings` with `/user/*` because it detects high cardinality on the second level, even though `/settings` is a static route.
+
+**Possible Solution using Weights**:
+
+This could be fixed by assigning a weight to each node, proportional to the number of transactions that have this segment.
+Nodes with large weights would then be encoded into the replacement rules as exceptions.
+
+We decided against this because encoding exceptions into rules would bloat project configs on the wire and in Relay.
+
+### False Positives
+
+The discovery of replacement rules is a best-effort approach: No matter how many rules the clusterer discovers, a project can always
+introduce a new feature that brings more high-cardinality transactions, and it takes time until the clusterer discovers a new rule for those.
+
+At the same time, the algorithm is blind to low-cardinality transactions that do not contain identifiers at all. For example, if a transaction
+like `/settings` has type `url`, neither the pattern-based nor the rule-based approach detect any identifiers.
+
+In order to prevent these false negatives, as of [this PR](https://github.com/getsentry/relay/pull/1960) we mark _every_ URL transaction as low-cardinality as long as there
+is _some_ scrubbing rule (even if it does not match), or we found an identifier pattern. In other words, we sacrifice [precision](https://en.wikipedia.org/wiki/Precision_and_recall) for the sake of [recall](https://en.wikipedia.org/wiki/Precision_and_recall).
+
+| category       | description                                                                          |
+| -------------- | ------------------------------------------------------------------------------------ |
+| true positive  | We scrubbed all identifiers (if any) and label the transaction as `sanitized`        |
+| false positive | We miss an identifier, but still label as `sanitized`                                |
+| true negative  | We keep the transaction labeled as `url` and it contains identifiers                 |
+| false negative | We keep the transaction labeled as `url` even though it does not contain identifiers |
```

Comment:
Not sure whether this table is helpful or confusing.

Reply:
I'm not sure if it helps, I'd probably remove it but your call. If you decide to leave it, I'd do it in a 2x2 contingency table, like [this one](https://www.google.com/url?sa=i&url=https%3A%2F%2Fwww.researchgate.net%2Ffigure%2FContingency-table-True-Positive-False-Positive-False-Negative-and-True-Negatives-are_fig5_280535795&psig=AOvVaw0DG6Iea5wQZxUOvnvRC5Mp&ust=1680340859920000&source=images&cd=vfe&ved=0CA8QjRxqFwoTCMjB6NHrhf4CFQAAAAAdAAAAABAE).

Reply:
Talked to @jan-auer and decided to keep it  as-is.